tests/unit/dccs/test_aws_iot.py

```diff
@@ -0,0 +1,615 @@
+        with self.assertRaises(Exception):
+            AWSIoT("Invalid object", enclose_metadata=True)
+
+    def test_implementation_register_entity(self):
+        """
+        Method to test the implementation of register method of AWSIoT for entity type.
+        :return: None
+        """
+
+        # Monkey patch the constructor of MqttDccComms
+        MqttDccComms.__init__ = mocked_init_mqtt_dcc_comms
+
+        # Creating the mocked MqttDccComms object
+        mocked_mqtt_dcc_comms = MqttDccComms()
+
+        aws = AWSIoT(mocked_mqtt_dcc_comms, enclose_metadata=True)
+
+        # Register the edge
+        registered_entity = aws.register(self.edge_system)
+
+        # Check the returned object is of the class RegisteredEntity
+        self.assertIsInstance(registered_entity, RegisteredEntity)
+
+    def test_implementation_register_metric(self):
+        """
+        Method to test the implementation of register method of AWSIoT for entity type.
+        :return: None
+        """
+
+        # Monkey patch the constructor of MqttDccComms
+        MqttDccComms.__init__ = mocked_init_mqtt_dcc_comms
+
+        # Creating the mocked MqttDccComms object
+        mocked_mqtt_dcc_comms = MqttDccComms()
+
+        aws = AWSIoT(mocked_mqtt_dcc_comms, enclose_metadata=True)
+
+        # Creating test Metric
+        test_metric = Metric(
+            name="Test_Metric",
+            unit=None,
+            interval=10,
+            aggregation_size=2,
+            sampling_function=sampling_function
+        )
+
+        registered_metric = aws.register(test_metric)
+
+        # Check the returned object is of the class RegisteredMetric
+        self.assertIsInstance(registered_metric, RegisteredMetric)
+
+    def test_implementation_create_relationship(self):
+        """
+        Method to test the implementation of create_relationship method of class AWSIoT.
+        :return: None
+        """
+
+        # Monkey patch the constructor of MqttDccComms
+        MqttDccComms.__init__ = mocked_init_mqtt_dcc_comms
+
+        # Creating the mocked MqttDccComms object
+        mocked_mqtt_dcc_comms = MqttDccComms()
+
+        aws = AWSIoT(mocked_mqtt_dcc_comms, enclose_metadata=True)
+
+        # Register the edge
+        registered_entity = aws.register(self.edge_system)
+
+        #  Creating test Metric
+        test_metric = Metric(
+            name="Test_Metric",
+            unit=None,
+            interval=10,
+            aggregation_size=2,
+            sampling_function=sampling_function
+        )
+
+        registered_metric = aws.register(test_metric)
+
+        # Creating the parent-child relationship
+        aws.create_relationship(registered_entity, registered_metric)
+
+        self.assertEqual(registered_metric.parent, registered_entity, "Check the implementation of create_relationship")
+
+    def test_validation_create_relationship_metric_device(self):
+        """
```

Comment:
Change comment to:
"Test case to check RegisteredMetric as Parent and RegisteredEntity as child"